lib/reverse_words.rb

```diff
@@ -1,4 +1,68 @@
 # A method to reverse each word in a sentence, in place.
+
+=begin
+think of finding:
+first index of the word
+last index of the word
+be mindful of finding the spaces before, after, and in between the words
+=end
+
+
 def reverse_words(my_words)
+  return nil if my_words == nil
+  return '' if my_words == ''
+
+  i = 0
+  all_words_end = my_words.length
+
+  while i < all_words_end
+
+    while my_words[i] == ' ' && i < all_words_end
+      i += 1
+    end
+
+    start_of_word = i
+
+    while my_words[i] != ' ' && i < all_words_end
+      i += 1
+    end
+
+    end_of_word = i - 1
+
+    reverse_string(my_words, start_of_word, end_of_word)
+  end
+
+  return my_words
```

Comment:
Line 56 and line 35 could simply be changed to `return` since you're making all the string manipulations in place.